### PR TITLE
Add producer and consumer resource constructors and a couple examples

### PR DESCRIPTION
**Pull request by amohrland**
Adds first example app, along with a couple of convenience methods.

### CODE REVIEW

File: examples/src/main/scala/example1/ExampleApp.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+      _ <- AdminApi.createTopicsIdempotent[F](kafkaBootstrapServers, topic)
+      _ <- Sync[F].delay(println(s"Created topic ${topic.name}"))
+
+      schemaRegistry <- SchemaRegistryApi(schemaRegistryUri)
+      _ <- schemaRegistry.registerKey[CustomerId](topic.name)
+      _ <- Sync[F].delay(println(s"Registered key schema for topic ${topic.name}"))
+
+      _ <- schemaRegistry.registerValue[Customer](topic.name)
+      _ <- Sync[F].delay(println(s"Registered value schema for topic ${topic.name}"))
```
Comment (zcox) [inline examples/src/main/scala/example1/ExampleApp.scala]: I bet we could make a nice single convenience function for registering schemas, similar to `createTopicsIdempotent`. Something like `SchemaRegistryApi.register[K, V](uri, topic)`.

File: examples/src/main/scala/example1/ExampleApp.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  val consumerResource = Resource.make(
+    ConsumerApi.avro4sShifting[F, CustomerId, Customer](
+      BootstrapServers(kafkaBootstrapServers),
+      SchemaRegistryUrl(schemaRegistryUri),
+      ClientId("consumer-example"),
+      GroupId("consumer-example-group"),
+      EnableAutoCommit(false)
+    )
+  )(_.close)
```
Comment (zcox) [inline examples/src/main/scala/example1/ExampleApp.scala]: I think you mentioned this before, but would also be nice to have convenience functions that returned `Resource[F, ProducerApi[F, K, V]]` and `Resource[F, ConsumerApi[F, K, V]]`.

File: examples/src/main/scala/example1/ZioApp.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+package example1
+
+import zio._
+import zio.interop.catz._
+
+object ZioApp extends CatsApp {
+  override def run(args: List[String]): ZIO[ZioApp.Environment, Nothing, Int] =
+    ExampleApp[Task].example.fold(_ => 1, _ => 0)
+}
```
Comment (zcox) [inline examples/src/main/scala/example1/ZioApp.scala]: Neat to see Zio usage in here!

File: examples/src/main/scala/example1/ExampleApp.scala (hunk 1 above)
Comment (amohrland) [reply examples/src/main/scala/example1/ExampleApp.scala]: Good idea, that sounds handy - I'll add that in.

File: examples/src/main/scala/example1/ExampleApp.scala (hunk 2 above)
Comment (amohrland) [reply examples/src/main/scala/example1/ExampleApp.scala]: Yeah, was considering adding those in this PR, but I think I'll add those in another PR, along with another example that uses the higher-level utils.

**Comment by amohrland** [comment]
I went ahead and added the convenience functions here and added a second example.

**Pull request merged**